Parsers/WatchGuardFirebox.txt

```diff
@@ -1,25 +1,43 @@
 // Title:           WatchGuard syslog Parser
 // Version:         1.0
-// Last Updated:    11/12/2020
+// Last Updated:    3/25/2021
 // Comment:         Inital Release
 //  
 // DESCRIPTION:
 // This parser extract some items from Syslog data and display the data in Azure Sentinel logs queries UI
 //
 // USAGE:
-// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. click Run.
-// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
-// It is recommended to name the Function Alias, as WatchGuardFirebox
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window.
+// 2. Click the Save->Save above the query. A Save pane will appear on the right, select "Function" from the "Save as" drop down. Enter a Function Name.
+// It is recommended to name the "Function Alias" as WatchGuardFirebox."Name" and "Category" also better to named WatchGuardFirebox.
 // 3. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
 //
 // REFERENCES: 
 // Using functions in Azure monitor log queries: https://docs.microsoft.com/azure/azure-monitor/log-query/functions
-
+let fromat_result = (source_arry:dynamic)
+{
+    let source_ips = array_concat(source_arry[0], source_arry[1]);
+    iif(source_ips[2]=="", strcat(source_ips[0],source_ips[1]), strcat(source_ips[0], dynamic(","),source_ips[2]))
+};
 Syslog
-| extend Interface = extract("msg_id=\"3100-002C\" \\[(.*)\\]",1,SyslogMessage),
-Intstatus = extract("Interface link status changed to ([\\w\\s]+)",1,SyslogMessage),
+| extend PolicyName = replace("-00","",extract(@"\s\(([-\w\s]*?(-00|\sPolicy|DVCP-BOVPN-Allow-in))\)$",1,SyslogMessage, typeof(string))),
+PolicyAction = extract("msg_id=\".*?\"\\s(\\w+?)\\s.*(Policy|-00|DVCP-BOVPN-Allow-in)\\)$",1,SyslogMessage),
+ProxyName = extract("Proxy.*?: ([\\w\\s]+)",1,SyslogMessage),
+Application = extract("app_name=\"(.*?)\"",1,SyslogMessage),
+MessageId = extract("msg_id=\"(.*?)\"",1,SyslogMessage),
+Message = extract("msg=\"(.*?)\"",1,SyslogMessage),

```

Comment:
Should it not be 'EventMessage'? Please refer https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema for field names.

Thanks,
Aditi Maheshwari